service/frontend/handler.go

```diff
@@ -66,9 +66,11 @@ type (
 	}
 
 	getHistoryContinuationToken struct {
-		RunID            string
-		NextEventID      int64
-		PersistenceToken []byte
+		RunID             string
+		FirstEventID      int64
```

Comment:
FirstEventID seems redundant.  The pagination token should already have enough information to continue where the user left off.

Reply:
getHistoryContinuationToken struct {
	RunID             string		
	NextEventID       int64
	IsWorkflowRunning bool // ignore this for now
	PersistenceToken  []byte
}

this is how the token will look like without the FirstEventID

think about the case:
pagination until next event ID is done in this API call (meaning PersistenceToken == nil) and there is no new event (meaning NextEventID does not change).  

if we return the this token to caller, there can be ambiguity when the next paging API call happens.
since the PersistenceToken == nil and nextEventID is a valid number, should the API beginning paging from event ID == 0 or should the API continue to page from nextEventID in the token?

Reply:
One more thing, query should not change during the pagination, which means first event ID should be stored.

Note that the paging state can only be reused with the exact same statement (same query string, same parameters).

https://docs.datastax.com/en/developer/java-driver/3.1/manual/paging/